### ISSUE TITLE
first_where false positive when using with Realm Result container

### ISSUE THREAD

**Issue by abagmut** [realm/SwiftLint#1930]
```swift
// This triggers a violation:
realm?.objects(User.self).filter(NSPredicate(format: "email ==[c] %@", email)).first
```

**Comment by jpsim** [comment]
Can you please fill out the issue template? It's hard to understand what you're actually reporting.

**Comment by abagmut** [comment]
`realm?.objects(User.self).filter(NSPredicate(format: "email ==[c] %@", email)).first`
this will produce warning

**Comment by jpsim** [comment]
Ok, please update the issue template with that so this issue can be understandable for contributors who land here.

**Comment by Dschee** [comment]
@abagmut I can totally understand what this is about and this should of course be fixed. I came across this, too, when using Realm `filter` functions, here's another example where this triggers but shouldn't:

```Swift
} else if let pauseToSplit = timeTracker.pauses.filter("beginDate < %@ AND endDate > %@", beginDate, endDate).first {
```

**Comment by sk-** [comment]
Note that the whitelist does not work in cases in which you call a function that returns an `NSPredicate`.

```swift
realm.objects(Model.self).filter(Model.predicate(x: x, y: y)).first
```

Also note that this issue should also be fixed for the `last_where` check.

@dschee could you reopen this issue, or should I file a new one?

**Comment by jpsim** [comment]
Please file a new issue to track that.

**Comment by jpsim** [comment]
We may choose to not fix it though since we won't be able to tell if the reference produces an NSPredicate or a closure.
